app/clients/zendesk_sell.py

```diff
@@ -55,21 +60,44 @@ def _generate_lead_data(contact: ContactRequest) -> Dict[str, Union[str, List[st
             }
         }
 
-    def send_contact_request(self, contact: ContactRequest) -> int:
-        ret = 200
-        if contact.is_demo_request():
-            ret = self.send_lead(contact)
+    @staticmethod
+    def _generate_contact_data(user: User) -> Dict[str, Union[str, List[str], Dict]]:
 
-        return ret
+        # validation based upon api mandatory fields
+        assert len(user.name) and len(user.email_address), 'Name or email are mandatory field'
+
+        first_name, last_name = ZenDeskSell._name_split(user.name)
+        return {
+            'data': {
+                'last_name': last_name,
+                'first_name': first_name,
+                'email': user.email_address,
+                'mobile': user.mobile_number,
+            }
+        }
+
+    @staticmethod
+    def _generate_deal_data(contact_id: int, service: Service, stage_id: int) -> Dict[str, Union[str, List[str], Dict]]:
+        return {
+            'data': {
+                'contact_id': contact_id,
+                'name': service.name,
+                'stage_id': stage_id,
+            }
```

Comment:
Could we have an external ID? Services can change their name.

Reply:
Yeah that's a good suggestion, I was really worried about the lifecycle of a deal. There are situations where if the user changes the service name that we wouldn't be able to track it within zendesk as the deal name wouldn't match.

Which then moves to another integration point, do we want to also send an update on [service name -> deal name] if the service name changes at any point in time? Currently to my knowledge we don't have that and the lifecycle of deal isn't tracked past the service creation.